### PR TITLE
i3: check model

### LINKED ISSUE #8: Add back model checks for the basic and carehome models

### PR DESCRIPTION
**Pull request by richfitz**
Ports the original checks from sircovid, verifying that models behave correctly in known edge cases.

No package code is changed in this PR: this just adds tests

There is one test that is slightly off; the first one in test-carehomes-test.R, where we check that the population is constant. The population *is* constant but due to rounding we don't quite get calculation exact people through the first step.

Fixes #8 (not sure why I have this branch named incorrectly!)

### CODE REVIEW

File: tests/testthat/test-carehomes-check.R [hunk 1]
```diff
@@ -0,0 +1,616 @@
...
+  info <- mod$info()
+  y0 <- carehomes_initial(info, 1, p)$state
+  mod$set_state(carehomes_initial(info, 1, p)$state)
+  mod$set_index(integer(0))
+  y <- mod$transform_variables(
+    drop(dust::dust_simulate(mod, seq(0, 400, by = 4))))
+
+  skip("This needs checking")
+
+  ## This is not quite correct, and I don't really know why. I think
```
Comment (lwhittles) [inline tests/testthat/test-carehomes-check.R:15]: I've tracked this back and you're right, the discrepancy occurs in 'carehomes_population' due to the rounding when removing residents/ workers from their respective groups. Maybe we should write a function in utils that fixes this? eg

```suggestion
round_int <- function(x) {
 x_floor <- floor(x)
 n_round_up <- n - sum(x_floor)
 i <- order(x - x_floor, decreasing = TRUE)[seq_len(n_round_up)]
 x_floor[i] <- x_floor[i] + 1
 x_floor
}
```
